src/ProgressOnderwijsUtils/ProcessRunner.cs

```diff
@@ -47,79 +48,72 @@ public Process CreateProcessObj()
+                    exitCodeCompletion.TrySetException(ex);
                 }
+                MarkOnePartClosed();
             };
 
-            Action fakeStdOutputEnd = null, fakeStdErrEnd = null;
-            var stdout = Observable.Create<(ProcessOutputKind Kind, string Content, TimeSpan Offset)>(observer => {
-                fakeStdOutputEnd = observer.OnCompleted;
-                proc.OutputDataReceived += (sender, e) => {
-                    if (e.Data == null) {
-                        observer.OnCompleted();
-                    } else {
-                        observer.OnNext((ProcessOutputKind.StdOutput, e.Data, stopwatch.Elapsed));
-                    }
-                };
-                return Disposable.Empty;
-            });
-            var stderr = Observable.Create<(ProcessOutputKind Kind, string Content, TimeSpan Offset)>(observer => {
-                fakeStdErrEnd = observer.OnCompleted;
-                proc.ErrorDataReceived += (sender, e) => {
-                    if (e.Data == null) {
-                        observer.OnCompleted();
-                    } else {
-                        observer.OnNext((ProcessOutputKind.StdError, e.Data, stopwatch.Elapsed));
-                    }
-                };
-                return Disposable.Empty;
-            });
+            var stdout = Observable.Create<(ProcessOutputKind Kind, string Content, TimeSpan Offset)>(
+                observer => {
+                    proc.OutputDataReceived += (sender, e) => {
+                        if (e.Data == null) {
+                            observer.OnCompleted();
+                            MarkOnePartClosed();
+                        } else {
+                            observer.OnNext((ProcessOutputKind.StdOutput, e.Data, stopwatch.Elapsed));
+                        }
+                    };
+                    return Disposable.Empty;
+                });
+            var stderr = Observable.Create<(ProcessOutputKind Kind, string Content, TimeSpan Offset)>(
+                observer => {
+                    proc.ErrorDataReceived += (sender, e) => {
+                        if (e.Data == null) {
+                            observer.OnCompleted();
+                            MarkOnePartClosed();
+                        } else {
+                            observer.OnNext((ProcessOutputKind.StdError, e.Data, stopwatch.Elapsed));
+                        }
+                    };
+                    return Disposable.Empty;
+                });
             var replayableMergedOutput = stdout.Merge(stderr).Replay();
             replayableMergedOutput.Connect();
             stopwatch.Start();
             proc.Start();
-            try {
-                proc.BeginErrorReadLine();
-            } catch {
-                //Beware: microsoft is utterly incompetent, so this code is in an intrinsic race condition with process exit, which you can emulate by sleeping before this try.

```

Comment:
Not accurate - the race is with dispose-on-exit; and that we can avoid by (potentially) disposing on another thread; whichever is last.